content/blog/2022/2022-08-30-handling-timezones-in-spring.md

```diff
@@ -0,0 +1,516 @@
+
+This applies to all databases, so conforming with the preferred practice, we will configure MySql to use UTC as default when working with JPA.
+This removes the complication of converting between timezones. Now we just need to handle timezones at the server.
+
+For this to apply, we will configure the below properties in `application.yml`
+````yaml
+spring:
+  jpa:
+    database-platform: org.hibernate.dialect.MySQL8Dialect
+    properties:
+      hibernate:
+        jdbc:
+          time_zone: UTC
+````
+
+### Mapping MySql datatypes to Java
+
+Let's take a quick look at some [MySql datatypes](https://dev.mysql.com/doc/refman/8.0/en/date-and-time-types.html).
+
+- **DATE** : The DATE type is used for values with a date part but no time part. MySQL retrieves and displays DATE values in 'YYYY-MM-DD' format.
+- **DATETIME** - The DATETIME type is used for values that contain both date and time parts. MySQL retrieves and displays DATETIME values in 'YYYY-MM-DD hh:mm:ss' format.
+- **TIMESTAMP** - The format for TIMESTAMP is similar to DATETIME. The only difference being TIMESTAMP by default stores values in UTC.
+- **TIME** - The TIME type stores the time in 'hh:mm:ss' format. But it can also store time up to microseconds (6 digits precision).
+
+Now that we understand the supported datatypes, let's look at how to map them with the Java Date/Time API.
+
+The MySql table is defined as follows:
+````sql
+    CREATE TABLE IF NOT EXISTS `timezonedb`.`date_time_tbl` (
+    `id`INT NOT NULL AUTO_INCREMENT,
+    `date_str` VARCHAR(500) NULL,
+    `date_time` DATETIME NULL,
+    `local_time` TIME NULL,
+    `local_date`  DATE NULL,
+    `local_datetime_dt` DATETIME NULL,
+    `offset_datetime` TIMESTAMP NULL,
+    `zoned_datetime` TIMESTAMP NULL,
+    `created_at` TIMESTAMP NOT NULL,
+
+    PRIMARY KEY (`id`));
+````
+
+The corresponding JPA entity is as below:
+````java
+    @Entity
+    @Table(name = "date_time_tbl")
+    public class DateTimeEntity implements Serializable {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Integer id;
+
+    @Column(name = "date_str")
+    private String dateStr;
+
+    @Column(name = "date_time")
+    private Date date;
+
+    @Column(name = "local_date")
+    private LocalDate localDate;
+
+    @Column(name = "local_time")
+    private LocalTime localTime;
+
+    @Column(name = "local_datetime_dt")
+    private LocalDateTime localDateTime;
+
+    @Column(name = "local_datetime_ts")
+    private LocalDateTime localDateTimeTs;
+
+    @Column(name = "offset_datetime")
+    private OffsetDateTime offsetDateTime;
+
+    @Column(name = "zoned_datetime")
+    private ZonedDateTime zonedDateTime;
+
+    @Column(name = "created_at", nullable = false, updatable = false)
+    @CreationTimestamp
+    private OffsetDateTime createdAt;
+
+````
+
+### Understanding the server timezone setup
+
+{{% image alt="settings" src="images/posts/handling-timezones-in-spring/TZ.JPG" %}}
```

Comment:
Please make this a table in the article (again, this is more readable and more inclusive for people with bad eyesight)